mrjob/py2.py

```diff
@@ -0,0 +1,170 @@
+
+Instead of using ``StringIO`` to deal with these, use ``io.BytesIO``.
+
+Not that both Python 2.6+ and Python 3.3+ have the ``bytes`` type and
+``b''`` constants built-in.
+
+These things should always be strings:
+
+- streams that you print() to (e.g. ``sys.stdout`` if you mock it out)
+- streams that you log to
+- paths on filesystem
+- URIs
+- arguments to commands
+- option names
+- Hadoop counter names and groups
+- Hadoop status messages
+- anything else we parse out of log files
+
+These things are strings because it makes for simpler code:
+
+- contents of config files
+- contents of scripts output by mrjob (e.g. the setup wrapper script)
+- contents of empty files
+
+Use the ``StringIO`` from this module to deal with strings (it's
+``StringIO.StringIO`` in Python 2 and ``io.StringIO`` in Python 3).
+
+Please use ``%`` for format strings and not ``format()``, which is much more
+picky about mixing unicode and bytes.
+
+This module provides a ``basestring`` type so you can check if something
+is a "string".
+
+We don't provide a ``unicode`` constant:
+
+- Use ``not isinstance(..., bytes)`` to check if a string is Unicode
+- To convert ``bytes`` to ``unicode``, use ``.decode('utf_8')`.
+- Python 3.3+ has ``u''`` literals; please use sparingly
+
+If you need to convert bytes of unknown encoding to a "string" (e.g.
+to ``print()`` or log them), you can use ``to_string()``.
+
+Iterables
+---------
+
+Using ``.iteritems()`` or ``.itervalues()`` in Python 2 to iterate over a
+dictionary when you don't need a list is best practice, but it's also (in most
+cases) an over-optimization. We'd prefer clean code; just use ``.items()``
+and ``.values()``.
+
+If you *do* every need that extra efficiency ``for k in some_dict`` does not
+create a list in either version of Python, and you for performance-critical
+code, it's fine to write custom code for each Python version (use ``IN_PY2``
+from this module to check which version you're in).
+
+Same goes for ``xrange``; just use ``range`` (this module provides ``xrange``
+solely to support ``ReprProtocol``).
+
+Miscellany
+----------
+
+We provide a ``long`` type (aliased to ``int`` on Python 3) so you can
+check if something is an integer: ``isinstance(..., (int, long))``.
+
+Any standard library function that deals with URLs (e.g. ``urlparse()``) should
+probably be imported from this module.
+
+You *usually* want to do ``from __future__ import print_function`` in modules
+where you use ``print()``. ``print(...)`` works fine, but
+``print(..., file=...)`` doesn't, and ``print()`` prints ``()`` on Python 2.
+
+You shouldn't need any other ``__future__`` imports.
+"""
+import sys
+
+# use this to check if we're in Python 2
+IN_PY2 = (sys.version_info[0] == 2)
+
+# ``basestring``, for ``isinstance(..., basestring)``
+if IN_PY2:
+    basestring = basestring
+else:
+    basestring = str
+basestring
+
+# ``long``, for ``isinstance(..., (int, long))``
+if IN_PY2:
+    long = long
+else:
+    long = int
+long
```

Comment:
Oh, I see. Yeah, I should at least use the same names as `six`

 - [ ] use `integer_types` instead of `(int, long)`
 - [ ] use `string_types` instead of `basestring`